### PR TITLE
Update default value of results from 0.0 to None to align with the default value in speckle-sharp

### PR DESCRIPTION
**Pull request by ItsPatrickHe**
## Description & motivation

I made this PR because in the speckle sharp repository, for example, as you can see for the Result1D class, the initial default value for all the attributes will be null when not set. These are indicated in the type as well. Could you check this PR to see if this change is correct?
image

## Checklist:

- [x] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: src/specklepy/objects/structural/results.py [hunk 1]
```diff
@@ -21,28 +21,28 @@ class ResultSet1D(Result, speckle_type=STRUCTURAL_RESULTS + "ResultSet1D"):
...
-    axialStress: float = 0.0
-    shearStressY: float = 0.0
-    shearStressZ: float = 0.0
-    bendingStressYPos: float = 0.0
-    bendingStressYNeg: float = 0.0
-    bendingStressZPos: float = 0.0
-    bendingStressZNeg: float = 0.0
-    combinedStressMax: float = 0.0
-    combinedStressMin: float = 0.0
+    position: float = None
```
Comment (gjedlicska) [inline src/specklepy/objects/structural/results.py:24]: In every place, where you say 
`{attribute_name}: float = None`
The actual definition should be `{attribute_name}: Optional[float] = None

If a `None | float` value is declared as just a `float` type, IDE-s and type checkers do not require you to handle the `None` case, which will most likely cause bugs. To be explicitly safe I strongly prefer using `Optional[float]` for these type definitions.

Please refactor them to that

**Review by gjedlicska — CHANGES_REQUESTED** [review]

File: src/specklepy/objects/structural/results.py (hunk 1 above)
Comment (ItsPatrickHe) [reply src/specklepy/objects/structural/results.py]: @gjedlicska Good one, I updated the type definition now.

**Review by gjedlicska — APPROVED** [review]

**Comment by RobClaessensRHDHV** [comment]
Hi @gjedlicska,
Great that his has been merged, together with your other updates of the specklepy structural objects (https://github.com/specklesystems/specklepy/pull/241). We are depending on these updates for our own repo.
Do you know when these updates will be released as pip installable?

**Comment by gjedlicska** [comment]
Hey @RobClaessensRHDHV 

We've just published a new release of specklepy -> `2.11.0`
There are a few bigger changes in this one, so please let me know if you run into anything unexpected.

**Comment by RobClaessensRHDHV** [comment]
@gjedlicska 

Nice! We will check it out next week :)

**Pull request merged**